### PR TITLE
Recode doctl sls init in pure golang

### PR DESCRIPTION
**Pull request by joshuaauerbachwatson**
This change replaces the call through the plugin to `nim project create` with pure golang code that does the equivalent.

### CODE REVIEW

**Review by ddebarros — CHANGES_REQUESTED** [review]
Hey Josh, I noticed when I run `doctl sls init --language=golang2` or any unknown language, a file `sample/hello.` gets created with the extension missing. 

Everything else looks good.

**Comment by joshuaauerbachwatson** [comment]
> when I run `doctl sls init --language=golang2` or any unknown language, a file `sample/hello.` gets created with the extension missing.

Indeed, there was a missing error check.   Adding the error check made the command behave better but subsequent failures of the unit tests showed that not all assumptions were actually correct.   The default language 'javascript' is applied by cobra as a side-effect of parsing, so the unit tests needed to take this into account.    In addition, when the runtime implied by the language is 'nodejs' the command will check for the presence of a `nodejs` runtime, which IMO is unnecessary and illogical (the runtime necessary to support the default language should be assumed to be available _a priori_).   Rather than revise the tests still further to always list a `nodejs` runtime, I gave that runtime name an automatic pass.

The PR is also now "rebased" to `main` rather than `feature/serverless` in keeping with our new practice.   This more or less lost all the previous commit history so it's probably worth doing another quick review to make sure I've covered all the bases.

File: commands/serverless-extra.go [hunk 1]
```diff
@@ -84,34 +86,78 @@ the contents to the cloud incrementally as it detects changes.`,
...
+	// Determine the kind and sample
+	kind, sample, ts, err := languageToKindAndSample(c, language)
 	if err != nil {
 		return err
 	}
-	output, err := RunServerlessExec(projectCreate, c, []string{flagOverwrite}, []string{flagLanguage})
+
+	// Make the config and various paths
+	config := configTemplate()
+	configFile := filepath.Join(project, "project.yml")
```
Comment (ddebarros) [inline commands/serverless-extra.go:104]: Should we support  extension `yaml` instead ? [That is the recommended extension](https://yaml.org/faq.html). Not a blocker.

**Review by ddebarros — DISMISSED** [review]
LGTM 👍🏾

File: commands/serverless-extra.go (hunk 1 above)
Comment (joshuaauerbachwatson) [reply commands/serverless-extra.go:104]: Agreed we should to it.  I think not in this PR, though.

File: commands/serverless-extra.go [hunk 2]
```diff
@@ -184,6 +243,159 @@ func RunServerlessExtraWatch(c *CmdConfig) error {
...
 }
 
+// prepareProjectArea prepares a disk area for receiving a project.  If the area exists and is not empty,
+// the overwrite flag must be true else error.  On successful return, the area either did not pre-exist
+// or has been removed.  Note: this function can be replaced for testing.
+var prepareProjectArea = func(project string, overwrite bool) error {
+	_, err := os.Stat(project)
+	if os.IsNotExist(err) {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (joshuaauerbachwatson) [reply commands/serverless-extra.go:253]: This should be addressed by a subsequent commit.

File: commands/serverless-samples.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
+/*
```
[the remark this answers is not in the document]
Comment (joshuaauerbachwatson) [reply commands/serverless-samples.go]: Done.

File: commands/serverless-samples.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+func Main(args map[string]interface{}) map[string]interface{} {
+\tname, ok := args["name"].(string)
+\tif !ok {
+\t\tname = "stranger"
+\t}
+\tmsg := make(map[string]interface{})
+\tmsg["body"] = "Hello " + name + "!"
+\treturn msg
+}
```
[the remark this answers is not in the document]
Comment (joshuaauerbachwatson) [reply commands/serverless-samples.go]: I changed the representation in the sample to literal tabs.  That seems to fix the problem.  I assume it's the correct style.

**Review by andrewsomething — APPROVED** [review]

**Pull request merged**